### PR TITLE
Update provider.go

### PR DESCRIPTION
**Pull request by windyear**
optimize: add PortForward to the Provider interface and add the PortForward from the provider to the PodHandlerConfig.

### CODE REVIEW

**Comment by aka-somix** [comment]
Hi @windyear !
Thank you for the PR. I saw the conversation on the upstream pull request. You are totally correct we forgot to add this part. 
Since the PR will be managed directly upstream, I wil close this one for now.

If anything changes we will reopen it from here. 🚀

**Review by aka-somix — APPROVED** [review]
LGTM

**Comment by aka-somix** [comment]
@windyear it looks like there is something missing. 
From the logs i see that the MockProvider is no longer implementing the Provider interface.  It should be because the Portforward method is not implemented. 

Could you please also provide a fast mock implementation for it?

Thanks 🚀

**Comment by windyear** [comment]
add fast mock implementation for mock provider with portforward](https://github.com/aka-somix/virtual-kubelet/pull/2/commits/a330865499c76ab80e01a3febfca1e32a9b64bbe). @aka-somix

**Comment by ssoBAekiL** [comment]
@windyear some changes are not passing the linting requirements. Could you take a look at that?
As soon as all tests are successful we'll merge the PR.

Thanks!

File: node/nodeutil/provider.go [hunk 1]
```diff
@@ -73,6 +76,7 @@ func AttachProviderRoutes(mux api.ServeMux) NodeOpt {
 				GetMetricsResource:    p.GetMetricsResource,
 				StreamIdleTimeout:     cfg.StreamIdleTimeout,
 				StreamCreationTimeout: cfg.StreamCreationTimeout,
+				PortForward:      p.PortForward,
```
Comment (ssoBAekiL) [inline node/nodeutil/provider.go]: Indentation allignment should fix the lint issue

**Review by ssoBAekiL — CHANGES_REQUESTED** [review]

File: node/nodeutil/provider.go (hunk 1 above)
Comment (windyear) [reply node/nodeutil/provider.go]: already fix, thanks for point out

**Review by ssoBAekiL — APPROVED** [review]

**Pull request merged**